src/OrchardCore.Modules/OrchardCore.Autoroute/Routing/AutoRouteTransformer.cs

```diff
@@ -20,7 +20,7 @@ public AutoRouteTransformer(IAutorouteEntries entries, IOptions<AutorouteOptions
 
         public override ValueTask<RouteValueDictionary> TransformAsync(HttpContext httpContext, RouteValueDictionary values)
         {
-            if (_entries.TryGetContentItemId(httpContext.Request.Path.Value.TrimEnd('/'), out var contentItemId))
+            if (_entries.TryGetContentItemId(httpContext.Request.Path.Value, out var contentItemId))
```

Comment:
IMHO I don't wanna change anything in other modules, so we can sure we didn't break anything else ;)

Reply:
It's safe, the trim is in the autorouteEntries service itself, and it fixes an issue.